raft/raft.go

```diff
@@ -663,6 +663,11 @@ func stepFollower(r *raft, m pb.Message) {
 }
 
 func (r *raft) handleAppendEntries(m pb.Message) {
+	if m.Index < r.Commit {
```

Comment:
we also need to take care of the compacted log case. appendLog will return false if the matched log has been compacted.